src/plugins/auto/src/plugin.cpp

```diff
@@ -549,10 +549,15 @@ std::list<DeviceInformation> MultiDeviceInferencePlugin::GetValidDevice(
             continue;
         }
         if (item.deviceName.find("GPU") == 0) {
-            auto& gpuUniqueName = item.uniqueName;
-            if (gpuUniqueName.find("iGPU") != std::string::npos) {
+            ov::Any deviceType;
+            if (IsNewAPI()) {
+                deviceType = GetCore()->GetMetric(item.deviceName, METRIC_KEY(DEVICE_TYPE)).as<ov::device::Type>();
+            } else {
+                deviceType = GetCore()->GetMetric(item.deviceName, METRIC_KEY(DEVICE_TYPE)).as<Metrics::DeviceType>();
+            }
```

Comment:
why still need to check new API, device type is mapped in properties already:

/** @cond INTERNAL */
inline std::ostream& operator<<(std::ostream& os, const InferenceEngine::Metrics::DeviceType& deviceType) {
    switch (deviceType) {
    case InferenceEngine::Metrics::DeviceType::discrete:
        os << "discrete";
        break;
    case InferenceEngine::Metrics::DeviceType::integrated:
        os << "integrated";
        break;
    default:
        os << "unknown";
        break;
    }

    return os;
}

Reply:
Okay.